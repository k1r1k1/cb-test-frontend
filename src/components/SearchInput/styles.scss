.search-input {
  position: relative;

  .search-list {
    position: absolute;
    width: 100%;
    border: 1px solid #d3d3d3;
    border-radius: 6px;
    max-height: 175px;
    margin-top: 2px;
    overflow-y: scroll;
    overflow-x: hidden;
    background: #fff;
  }

}