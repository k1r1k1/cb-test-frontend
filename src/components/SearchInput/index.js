import React, { useState } from 'react'
import './styles.scss'

const SearchInput = ({
  items, isInvalid, setFieldValue, name,
  getItemString, getItemValue
}) => {
  const [foundItems, setItems] = useState()
  const [inputFocused, setFocused] = useState(false)
  const [inputState, setInputState] = useState('')

  const handleChange = (({ target }) => {
    setInputState(target.value)
    setFieldValue(name, null)
    if (target.value.length < 1) return setItems(null)
    setItems(
      items?.filter(item =>
        JSON.stringify(
          Object.values(item)
        )
          .toLowerCase()
          .indexOf(target.value?.toLowerCase()) !== -1
      )
    )
  })

  const handleItemClick = (item) => {
    setInputState(getItemString(item))
    setFieldValue(name, getItemValue(item))
  }

  return (
    <div className="search-input">
      <input
        className={`form-control ${isInvalid ? 'is-invalid' : ''}`}
        type="text"
        value={inputState}
        onChange={handleChange}
        onFocus={() => setFocused(true)}
        onBlur={() => setTimeout(() => setFocused(false), 80)}
      />
      {(inputFocused && foundItems?.length ?
        (<div className="search-list">
          <table className="table table-striped">
            <tbody>
              {foundItems.map((item, index) => (
                <tr
                  key={`search-input-value-${index}`}
                  onMouseDown={() => handleItemClick(item)}
                >
                  <td>
                    {getItemString(item)}
                  </td>
                </tr>
              ))}
            </tbody>
          </table>
        </div>) : '')}
    </div>
  )
}

export default SearchInput
