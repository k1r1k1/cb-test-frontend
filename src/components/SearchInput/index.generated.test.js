import renderer from 'react-test-renderer';
import React from 'react'

import SearchInput from './index';

jest.mock('./styles.scss');

const renderTree = tree => renderer.create(tree);
describe('<SearchInput>', () => {
  it('should render component', () => {
    expect(renderTree(<SearchInput 
    />).toJSON()).toMatchSnapshot();
  });
  
});