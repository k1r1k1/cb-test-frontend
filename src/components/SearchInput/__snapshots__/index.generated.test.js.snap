// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput> should render component 1`] = `
<div
  className="search-input"
>
  <input
    className="form-control "
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
    value=""
  />
</div>
`;
